install_script.sh.template

```diff
@@ -555,6 +555,40 @@ function install_installer() {
     return $exit_status
 }
 
+function is_installed_by_installer() {
+    local sudo_cmd="$1"
+    if command -v datadog-installer >/dev/null 2>&1; then
+        $sudo_cmd datadog-installer is-installed "$2"
+        local status=$?
+        if [ $status -eq 0 ]; then
+            return 0
+        elif [ $status -eq 10 ]; then # 10 means the package is not installed
+            return 1
+        else
+            return $status
+        fi
+    else
+        return 1
+    fi
+}
+
+function filter_packages_installed_by_installer() {
+    local sudo_cmd="$1"
+    local pkg_array_name=$2
+    local -a not_installed_packages=()  # Temporary array for not installed packages
+
+    for package in $(eval echo "\${${pkg_array_name}[@]}"); do
+        if ! is_installed_by_installer "$sudo_cmd" "$package"; then
+            not_installed_packages+=("$package")
+        else
+            echo -e "\033[34m\n* Skipping installation of $package as it is already installed by the installer\n\033[0m"
+        fi
+    done
+
+    # Copy the filtered array back to the original array
+    eval "$pkg_array_name=(\"\${not_installed_packages[@]}\")"
```

Comment:
I tried without eval using [nameref `-n` but it's only for bash 4.3+](https://www.gnu.org/software/bash/manual/bash.html#Locale-Translation).
Which makes me realise that we need to be sure that the array syntaxes are compatible across bash versions

Reply:
Yeah I tried nameref at first but it failed on Centos7 and other old platforms